### PR TITLE
fix(ilp): allow - in column name (#822)

### PR DESCRIPTION
**Pull request by zikani03**

### CODE REVIEW

File: core/src/main/java/io/questdb/cairo/TableUtils.java [hunk 1]
```diff
@@ -283,6 +283,44 @@ public static boolean isValidColumnName(CharSequence seq) {
         return true;
     }
 
+    public static boolean isValidInfluxColumnName(CharSequence seq) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/TableUtils.java:286]: In my benchmark:

https://gist.github.com/bluestreak01/e39374cc9b06749af747afe1f33547bb

This version is a little slower than the original:

```
Benchmark                                       Mode  Cnt   Score   Error  Units
IsValidColumnBenchmark.testIsValidColumn        avgt   40  18.928 ± 3.004  ns/op
IsValidColumnBenchmark.testIsValidInfluxColumn  avgt   40  20.833 ± 0.048  ns/op
```

I feel there is an optimization opportunity here:

```java
    public static boolean isValidInfluxColumnName(CharSequence seq) {
        for (int i = 0, l = seq.length(); i < l; i++) {
            switch (seq.charAt(i)) {
                default:
                    break;
                case ' ':
                case '?':
                case '.':
                case ',':
                case '\'':
                case '\"':
                case '\\':
                case '/':
                case '\0':
                case ':':
                case ')':
                case '(':
                case '+':
                case '*':
                case '%':
                case '~':
                case 0xfeff: // UTF-8 BOM (Byte Order Mark) can appear at the beginning of a character stream
                    return false;
                case '_':
                    if (i < 1) {
                        return false;
                    }
                    break;
                case '-':
                    if (i == 0 || i == l - 1) {
                        return false;
                    }
                    break;
            }
        }
        return true;
    }
```

The benchmark of this version on my laptop, with fluctuating C-states, looks like this:

```
Benchmark                                       Mode  Cnt   Score   Error  Units
IsValidColumnBenchmark.testIsValidColumn        avgt   40  18.940 ± 3.000  ns/op
IsValidColumnBenchmark.testIsValidInfluxColumn  avgt   40  14.080 ± 0.186  ns/op
```

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/cairo/TableUtils.java (hunk 1 above)
Comment (zikani03) [reply core/src/main/java/io/questdb/cairo/TableUtils.java:286]: Interesting! Thanks for pointing that out, will make the changes

**Review by bluestreak01 — APPROVED** [review]

**Comment by bluestreak01** [comment]
hey @allcontributors, please add @zikani03 for their awesome PR!

**Comment by bsmth** [comment]
@all-contributors bot, add @zikani03 for code, please!

**Pull request merged**